_deploy/r/gnoswap/gnft/gnft_test.gno

```diff
@@ -0,0 +1,450 @@
+package gnft
+
+import (
+	"std"
+	"testing"
+
+	"gno.land/p/demo/grc/grc721"
+	"gno.land/p/demo/testutils"
+	"gno.land/p/demo/uassert"
+
+	"gno.land/r/gnoswap/v1/consts"
+)
+
+const (
+	errInvalidTokenId = "invalid token id"
+)
+
+var (
+	positionAddr  = consts.POSITION_ADDR
+	positionRealm = std.NewCodeRealm(consts.POSITION_PATH)
+
+	addr01      = testutils.TestAddress("addr01")
+	addr01Realm = std.NewUserRealm(addr01)
+
+	addr02      = testutils.TestAddress("addr02")
+	addr02Realm = std.NewUserRealm(addr02)
+)
+
+func TestMetadata(t *testing.T) {
+	tests := []struct {
+		name     string
+		fn       func() string
+		expected string
+	}{
+		{"Name()", Name, "GNOSWAP NFT"},
+		{"Symbol()", Symbol, "GNFT"},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			uassert.Equal(t, tt.expected, tt.fn())
+		})
+	}
+}
+
+func TestTotalSupply(t *testing.T) {
+	tests := []struct {
+		name     string
+		setup    func()
+		expected uint64
+	}{
+		{
+			name:     "initial total supply",
+			expected: uint64(0),
+		},
+		{
+			name: "total supply after minting",
+			setup: func() {
+				std.TestSetRealm(positionRealm)
+				Mint(a2u(addr01), tid(1))
+			},
```

Comment:
A TC should be added that causes totalSupply to fluctuate on deletion via Burn.

Reply:
2e25c41db0c6c37365eed4918f2ad05521b8c61f